contracts/prover/MessageBridgeProver.sol

```diff
@@ -103,30 +102,44 @@ abstract contract MessageBridgeProver is
             revert UnauthorizedIncomingProof(messageSender);
         }
 
-        // Process the intent proofs directly from bytes data
-        // The source chain ID becomes the destination chain ID in the proof
-        _processIntentProofs(message, sourceChainId);
+        // Extract the chain ID from the beginning of the message
+        // Message format: [chainId (8 bytes as uint64)] + [encodedProofs]
+        if (message.length < 8) {
+            revert InvalidProofMessage();
+        }
+        
+        // Convert raw 8 bytes to uint64 - the chain ID is stored as big-endian bytes
+        bytes8 chainIdBytes = bytes8(message[0:8]);
+        uint256 actualChainId = uint256(uint64(chainIdBytes));
+        bytes calldata encodedProofs = message[8:];
+
+        // Process the intent proofs using the chain ID extracted from the message
+        _processIntentProofs(encodedProofs, actualChainId);
```

Comment:
This function should not take `actualChainId` at `uint256` as it casts it to `uint64` internally. Change the parameter type to avoid the cast back-and-forth.